GoogleForCreators/web-stories-wp
# Video Optimization: Respect user setting

## Context



## Summary

A new user provide to load data in from the current user, so user can opt / in out of video transcoding. 

## Relevant Technical Choices

- Add new user provider, to enable easier implementation of https://github.com/google/web-stories-wp/issues/6087
- `canTranscodeFile` becomes an async function, as we have to wait for the api request to get current user data. 
- Current user data is preloaded in edit, so we shouldn't have to wait around for the api response. 

## To-do



## User-facing changes

Checkbox in the settings screen now works, and if unchecks, the video optimization does not fire. 

## Testing Instructions

1. Enable video optimization in experiments 
2. Dashboard > settings 
3. Uncheck video optimization. 
4. Drag a .mov file in dashboard, see the file get rejected. 

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #6092
